### DISCUSSION TITLE
Generate Tessellation on sphere surface only

### DISCUSSION THREAD

**Discussion by woongy94** [neperfepx/neper#650] [Q&A] [answered]
Dear Romain

Thank you for providing such an amazing tool

I already made my -T script like below

neper -T -n from_morpho -morpho "diameq:lognormal(2,0.05),1-sphericity:lognormal(0.05,0.03)" -domain "sphere(10,1000)" -dim 3 -format "stl"

and it works perfectly :)

Now I want to generate a Tessellation only on the surface of the domain (in my work, it is sphere), and to connect the intersections of Tessellation with the center of the sphere by a line. Can you tell me is it possible or not?

Kind regards,
Jinwoong Choi

**Comment by rquey** [comment, marked as answer]
Your configuration is what you get when all the seeds are located on the sphere. You can give it a try by defining such seeds and loading their coordinates using `-morphooptiini "coo:file(...)"`.
The coordinates will have to be "randomized" a bit (not all exactly on the sphere).

Can I ask what your application is? Why are you building a 2D tessellation on the sphere?

**Reply by woongy94** [reply]
Thank you for your kind reply! :)

I'm applying it to make a battery cathode particle model.
I also success to make another kind of geometry which will applying to my another work.
But now I want to make this kind of structure using with Neper.
(Kind of centroidal voronoi tessellation? volumetric voronoi tessellation?...)

![1234](https://user-images.githubusercontent.com/121857641/222325345-b05690dd-d504-4dcc-ad23-b499409b858b.PNG)
(Figure from here : https://www.iaacblog.com/programs/fragmented-surfaces/)

So I was trying to make a 2D tessellation on the sphere..! :)


Jinwoong Choi.

**Reply by woongy94** [reply]
Oh, I got the solution from #149 ..

My approach was wrong I think..
and when I change my mind on my approach, I can find another solution to make my model.

Now I will going to understand and apply the 'centroid' in the Neper

I really appreciate you to supply Neper this is amazing tool :)


Thank you

Jinwoong Choi.
